core/orchestrator_core.go

```diff
@@ -3653,6 +4326,17 @@ func (o *TridentOrchestrator) ListSnapshotsForVolume(
 	return snapshots, nil
 }
 
+// ReadSnapshotsForVolume returns a list of snapshots for the specified volume
```

Comment:
Here is another case where deeper insight would help.  The previous method ```ListSnapshotsForVolume()``` returns all snapshots cached here in the orchestrator core, while ```ReadSnapshotsForVolume()``` invokes storage APIs to get the list of snapshots directly from a storage backend.  The comment here isn't wrong, but it could be much more helpful.